### PR TITLE
JIT: Unify LSRA call arg building

### PR DESCRIPTION
**Pull request by jakobbotsch**
This unifies the building of uses of call args across all our targets, and removes accesses of the old ABI information.

LSRA had a bunch of assert code to compare the `PUTARG_REG` nodes to the ABI info. I don't think this is particularly valuable given that we create `PUTARG_REG` nodes in one centralized place in lowering now that follows the same pattern as the new unified code here does.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @kunalspathak

**Comment by jakobbotsch** [comment]
Ping @kunalspathak

File: src/coreclr/jit/lsrabuild.cpp [hunk 1]
```diff
@@ -3149,7 +3149,85 @@ RefPosition* LinearScan::BuildDef(GenTree* tree, SingleTypeRegSet dstCandidates,
...
+    // need to remain live until the call.  In fact, it must not
+    // because the code generator doesn't actually consider it live,
+    // so it can't be spilled.
+
+    for (CallArg& arg : call->gtArgs.EarlyArgs())
+    {
+        assert(arg.GetEarlyNode()->OperIs(GT_PUTARG_STK));
+        assert(arg.GetLateNode() == nullptr);
+    }
+#endif // DEBUG
```
Comment (kunalspathak) [inline src/coreclr/jit/lsrabuild.cpp:3224]: are we missing this existing validation?

https://github.com/dotnet/runtime/blob/31733b9a35185785427bac69ef80a4eb56b727c2/src/coreclr/jit/lsraxarch.cpp#L1295-L1314

Comment (jakobbotsch) [reply src/coreclr/jit/lsrabuild.cpp:3224]: I don't think the validation is useful. I wrote a comment about it in the issue text above:

> LSRA had a bunch of assert code to compare the PUTARG_REG nodes to the ABI info. I don't think this is particularly valuable given that we create PUTARG_REG nodes in one centralized place in lowering now that follows the same pattern as the new unified code here does.

This validation was more useful before the new ABI info when the correspondence to it was super complicated. With the new ABI info it is much simpler.

The invariant here is that the code that created the `PUTARG_REG` filled out the ABI registers on those nodes. That code already has to query the ABI info to do that, so duplicating it in LSRA gets us nothing (then we might as well just remove it from lowering instead).

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**